classes/requests/order_management/class-truelayer-request-refunds.php

```diff
@@ -18,80 +21,78 @@ class TrueLayer_Request_Refunds extends TrueLayer_Request_Post {
+	 * @var WC_Order
 	 */
-	public function __construct( $arguments ) {
-		parent::__construct( $arguments );
-		$this->order_id        = $arguments['order_id'];
-		$this->order           = wc_get_order( $this->order_id );
-		$this->log_title       = 'Refund payment';
-		$this->endpoint        = '/payments/' . $this->order->get_transaction_id() . '/refunds';
-		$this->idempotency_key = Truelayer_Helper_Signing::get_uuid();
-
-		$this->amount = round( $arguments['amount'] * 100 );
-		$this->reason = $arguments['reason'];
-	}
+	public $order;
 
 	/**
-	 * Create the request body.
+	 * The amount to be refunded.
 	 *
-	 * @return string
+	 * @var int
 	 */
-	protected function get_body() {
+	public $amount;
 
-		$body = array(
-			'amount_in_minor' => $this->amount,
-			'reference'       => $this->reason,
-		);
-
-		return $body;
-	}
+	/**
+	 * The refund reason.
+	 *
+	 * @var string
+	 */
+	public $reason;
 
 	/**
-	 * Get the request url.
+	 * Class constructor.
 	 *
-	 * @return string
+	 * @param array $arguments the class constructor arguments array.
 	 */
-	protected function get_request_url() {
-		$truelayer_token_url = $this->get_api_url_base() . $this->endpoint;
+	public function __construct( $arguments ) {
+		parent::__construct( $arguments );
+		$this->order_id  = $arguments['order_id'];
+		$this->order     = wc_get_order( $this->order_id );
+		$this->log_title = 'Refund payment';
 
-		return $truelayer_token_url;
+		$this->amount = intval( round( $arguments['amount'] * 100 ) );
+		$this->reason = $arguments['reason'] ?? '';
 	}
 
 	/**
-	 * Request headers.
+	 * Make the request.
 	 *
-	 * @param array $body The request body.
-	 * @return array
+	 * @return RefundCreatedInterface|WP_Error
 	 */
-	protected function get_request_headers( $body = array() ) {
-		$token = TrueLayer()->api->get_token();
+	public function request() {
+		$this->client = $this->get_client();
 
-		return array(
-			'Content-Type'    => 'application/json',
-			'Idempotency-Key' => $this->idempotency_key,
-			'Tl-Signature'    => Truelayer_Helper_Signing::get_tl_signature( $body, $this ),
-			'TL-Agent'        => 'truelayer-woocommerce/' . TRUELAYER_WC_PLUGIN_VERSION,
-			'Authorization'   => "Bearer $token",
-		);
+		try {
+			return $this->refund_payment();
+		} catch ( Exception $e ) {
+			return new WP_Error( 'tl_refund_payment_error', $e->getMessage() );
+		}
 	}
 
 	/**
-	 * Get request arguments and check request body.
+	 * Refund the payment.
+	 *
+	 * @return RefundCreatedInterface|RefundFailedInterface|WP_Error
 	 *
-	 * @return bool|array
+	 * @throws Exception When the payment is not found.
 	 */
-	protected function get_request_args() {
-		$body = $this->get_body();
+	private function refund_payment() {
+		$payment_id = $this->order->get_transaction_id();
+		$reference  = ! empty( $this->reason ) ? $this->reason : __( 'Refund for order ', 'truelayer-for-woocommerce' ) . $this->order->get_order_number();
```

Comment:
May I suggest using `sprintf` here, and include the order number as `%s` (together with the translator comment)? This would make the intended purpose of the text clearer for the translator.